uni/lib/view/library/widgets/library_reservations_card.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/model/entities/library_reservation.dart';
+import 'package:uni/model/providers/library_reservations_provider.dart';
+import 'package:uni/model/request_status.dart';
+import 'package:uni/utils/drawer_items.dart';
+import 'package:uni/view/common_widgets/generic_card.dart';
+import 'package:uni/view/library/widgets/reservation_row.dart';
+
+class LibraryReservationsCard extends GenericCard {
+  LibraryReservationsCard({Key? key}) : super(key: key);
+
+  const LibraryReservationsCard.fromEditingInformation(
+      Key key, bool editingMode, Function()? onDelete)
+      : super.fromEditingInformation(key, editingMode, onDelete);
+
+  @override
+  onClick(BuildContext context) => Navigator.pushNamed(
+      context, '/${DrawerItem.navLibraryReservations.title}');
+
+  @override
+  String getTitle() => 'Gabinetes Reservados';
+
+  @override
+  Widget buildCardContent(BuildContext context) {
+    return Consumer<LibraryReservationsProvider> (
+      builder: (context, reservationsProvider, _) {
+      if (reservationsProvider.status == RequestStatus.busy) {
+        return const Center(child: CircularProgressIndicator());
+      } else {
+        return RoomsList(reservationsProvider.reservations);
+      }
+      });
+  }
+}
+
+class RoomsList extends StatelessWidget {
+  final List<LibraryReservation>? reservations;
```

Comment:
Does this need to be nullable, since the provider already returns an empty list when null?

Reply:
:+1: fixed